tor/core/inbox.py

```diff
@@ -157,50 +148,55 @@ def check_inbox(cfg):
     for item in reversed(list(cfg.r.inbox.unread(limit=None))):
         # Very rarely we may actually get a message from Reddit itself.
         # In this case, there will be no author attribute.
-        if item.author is None:
+        author_name = item.author.name if item.author else None
+        if author_name is None:
             send_to_modchat(
                 f'We received a message without an author -- '
                 f'*{item.subject}*:\n{item.body}', cfg
             )
             item.mark_read()
 
-        elif item.author.name == 'transcribot':
+        elif author_name == 'transcribot':
             item.mark_read()
```

Comment:
We can return here and just skip the rest of the processing since we know that we aren't going to do anything else to this comment.

Reply:
There's an argument that we could just add the transcribot user to the blacklist and not even need this at all too.

Realistically, the `if`-`elif`-`else` tree makes it so no other condition is evaluated already. Returning here (though I think you mean `continue` since it's a for-loop) would be redundant.

Reply:
You could move `item.mark_read()` to the end of the function, since every branch calls it anyway. Then replace this one with just `pass`.

I also like the idea of blacklisting transcribot but that would probably need to be documented somewhere.